docs/developer/development/ci.rst

```diff
@@ -0,0 +1,147 @@
+Continuous Testing
+==================
+
+Add a new test in the continuous integration system
+---------------------------------------------------
+
+A test as continuous integration system level is an End-to-End test that need a
+dedicated environment (with one or several machines (virtual or container)).
+
+A test that only check a specific feature of a classic MetalK8s deployment
+should be part of PyTest BDD and not integrated as a dedicated stage in
+continuous integration system.
+
+How to choose between Pre-merge and Post-merge
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Choice really depend on the goal of this test.
+
+As high level view:
+
+Pre-merge:
+
+- Test is not too long (less than 30 minutes).
```

Comment:
`Test is usually not long and could last less than 30 minutes` or
`Pre-merge test are usually not long and could last less than 30 minutes`